### ISSUE TITLE
perlop shouldn't say that C has a defined or

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11346]
Migrated from [rt.perl.org#90648](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=90648) (status was 'resolved')

Searchable as RT90648$

**Comment by p5pRT** [comment]
### From tchrist@perl.com

Perlop has this entry&#8203;:

&nbsp;    =head2 C\-style Logical Defined\-Or

That's wrong\.  It should be only&#8203;:

&nbsp;    =head2 Logical Defined\-Or

Because there is no such thing as a defined\-or in C\.

Sheesh\.

\-\-tom

**Comment by p5pRT** [comment]
### From @tux

On Sun\, 15 May 2011 14&#8203;:44&#8203;:09 \-0700\, tchrist1 \(via RT\)
\<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:

> \# New Ticket Created by  tchrist1 
> \# Please include the string&#8203;:  \[perl \#90648\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=90648 >
> 
> 
> Perlop has this entry&#8203;:
> 
>     =head2 C\-style Logical Defined\-Or
> 
> That's wrong\.  It should be only&#8203;:
> 
>     =head2 Logical Defined\-Or
> 
> Because there is no such thing as a defined\-or in C\.

As you are digging through all docs\,
In that area\, did you see my post last week&#8203;:

It was not /my/ decision to remove "err" \(or "dor"\)
There are however always people asking for it\, and when told that there
isn't any\, there are no docs referring to the reason why not\. Should
there be?

http&#8203;://www\.perlmonks\.org/?node\_id=660524

<details><summary>Inline Patch</summary>

```diff
diff --git a/pod/perlop.pod b/pod/perlop.pod
index c97c331..289c421 100644
--- a/pod/perlop.pod
+++ b/pod/perlop.pod
@@ -911,9 +911,9 @@ expressions.  It's equivalent to && except for the very low
 precedence.  This means that it short-circuits: i.e., the right
 expression is evaluated only if the left expression is true.

-=head2 Logical or, Defined or, and Exclusive Or
+=head2 Logical or and Exclusive Or
 X<operator, logical, or> X<operator, logical, xor>
-X<operator, logical, defined or> X<operator, logical, exclusive or>
+X<operator, logical, exclusive or>
 X<or> X<xor>

 Binary "or" returns the logical disjunction of the two surrounding
@@ -942,6 +942,8 @@ Then again, you could always use parentheses.
 Binary "xor" returns the exclusive-OR of the two surrounding expressions.
 It cannot short circuit, of course.

+There is no low precedence operator for defined-OR.
+
 =head2 C Operators Missing From Perl
 X<operator, missing from perl> X<&> X<*>
 X<typecasting> X<(TYPE)>


-- 
```
</details>

H\.Merijn Brand  http&#8203;://tux\.nl      Perl Monger  http&#8203;://amsterdam\.pm\.org/
using 5\.00307 through 5\.12 and porting perl5\.13\.x on HP\-UX 10\.20\, 11\.00\,
11\.11\, 11\.23 and 11\.31\, OpenSuSE 10\.1\, 11\.0 \.\. 11\.3 and AIX 5\.2 and 5\.3\.
http&#8203;://mirrors\.develooper\.com/hpux/           http&#8203;://www\.test\-smoke\.org/
http&#8203;://qa\.perl\.org      http&#8203;://www\.goldmark\.org/jeff/stupid\-disclaimers/

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From tchrist@perl.com

Yes\, I remember that\.

I suppose it's ok to say there is no corresponding low\-precedence version\.

\-\-tom

**Comment by p5pRT** [comment]
### From @cpansprout

On Sun May 15 23&#8203;:02&#8203;:40 2011\, hmbrand wrote&#8203;:
> On Sun\, 15 May 2011 14&#8203;:44&#8203;:09 \-0700\, tchrist1 \(via RT\)
> \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> 
> > \# New Ticket Created by  tchrist1
> > \# Please include the string&#8203;:  \[perl \#90648\]
> > \# in the subject line of all future correspondence about this issue\.
> > \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=90648 >
> >
> >
> > Perlop has this entry&#8203;:
> >
> >     =head2 C\-style Logical Defined\-Or
> >
> > That's wrong\.  It should be only&#8203;:
> >
> >     =head2 Logical Defined\-Or
> >
> > Because there is no such thing as a defined\-or in C\.

Fixed in 26d9d83b86\.

> As you are digging through all docs\,
> In that area\, did you see my post last week&#8203;:
> 
> It was not /my/ decision to remove "err" \(or "dor"\)
> There are however always people asking for it\, and when told that
> there
> isn't any\, there are no docs referring to the reason why not\. Should
> there be?
> 
> http&#8203;://www\.perlmonks\.org/?node\_id=660524
> 
> diff \-\-git a/pod/perlop\.pod b/pod/perlop\.pod
> index c97c331\.\.289c421 100644
> \-\-\- a/pod/perlop\.pod
> \+\+\+ b/pod/perlop\.pod
> @&#8203;@&#8203; \-911\,9 \+911\,9 @&#8203;@&#8203; expressions\.  It's equivalent to && except for the
> very low
>  precedence\.  This means that it short\-circuits&#8203;: i\.e\.\, the right
>  expression is evaluated only if the left expression is true\.
> 
> \-=head2 Logical or\, Defined or\, and Exclusive Or
> \+=head2 Logical or and Exclusive Or
>  X\<operator\, logical\, or> X\<operator\, logical\, xor>
> \-X\<operator\, logical\, defined or> X\<operator\, logical\, exclusive or>
> \+X\<operator\, logical\, exclusive or>
>  X\<or> X\<xor>
> 
>  Binary "or" returns the logical disjunction of the two surrounding
> @&#8203;@&#8203; \-942\,6 \+942\,8 @&#8203;@&#8203; Then again\, you could always use parentheses\.
>  Binary "xor" returns the exclusive\-OR of the two surrounding
> expressions\.
>  It cannot short circuit\, of course\.
> 
> \+There is no low precedence operator for defined\-OR\.
> \+
>  =head2 C Operators Missing From Perl
>  X\<operator\, missing from perl> X\<&> X\<\*>
>  X\<typecasting> X\<\(TYPE\)>

Applied as 59ab9d6e\.  Thank you\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
